src/middlewared/middlewared/apidocs/templates/websocket/jobs.md

```diff
@@ -0,0 +1,119 @@
+## Jobs
+
+Tasks which require time to execute are tagged as jobs.
+When client connects to an endpoint marked as a job, he/she receives a job id
+from the end point. With this job id, the client can query the status of the job
+to see the progress and status of job. If any errors arise, they are reflected
+in the output, else the output reflects the result being returned by the end point
+on it's completion.
+
+e.g. `ws://freenas.domain/websocket`
+
+### Example of connecting to end point marked as a job
+
+#### Client connects to websocket endpoint and sends a `connect` message.
+
+    :::javascript
+    {
+        "id": "6841f242-840a-11e6-a437-00e04d680384",
+        "msg": "method",
+        "method": "jail.start",
+        "params": ["jail_name"]
+    }
+
+#### Server answers with `job_id`.
+
+    :::javascript
+    {
+      "msg": "result",
+      "id": "c0bb5952-fc60-232a-3d6c-a47961b771a5",
+      "result": 53
+    }
+
+### Query Job Status
+
+Job status can be queried with `core.get_jobs` method.
+
+Request:
+
+    :::javascript
+    {
+      "id": "d8e715be-6bc7-11e6-8c28-00e04d680384",
+      "msg": "method",
+      "method": "core.get_jobs",
+      "params": [[["id", "=", 53]]]
+    }
+
+Response:
+
+    :::javascript
+    {
+      "id": "d8e715be-6bc7-11e6-8c28-00e04d680384",
+      "msg": "result",
+      "result": [{'id': 53, 'method': 'jail.start', 'arguments': ['abc'], 'logs_path': None, 'logs_excerpt': None, 'progress': {'percent': None, 'description': None, 'extra': None}, 'result': True, 'error': None, 'exception': None, 'exc_info': None, 'state': 'SUCCESS', 'time_started': datetime.datetime(2019, 10, 17, 0, 4, 4, 588000, tzinfo=datetime.timezone.utc), 'time_finished': datetime.datetime(2019, 10, 17, 0, 4, 10, 28000, tzinfo=datetime.timezone.utc)}]
+    }
+
```

Comment:
We could have a section with an example on how to subscribe to jobs and what it looks like.